src/hotspot/os/posix/malloctrace/mallocTrace2.hpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2023 SAP SE. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef OS_POSIX_MALLOCTRACE_MALLOCTRACE_HPP
+#define OS_POSIX_MALLOCTRACE_MALLOCTRACE_HPP
+
+#include "services/diagnosticCommand.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+#if defined(LINUX) || defined(__APPLE__)
+
+class outputStream;
+
+namespace sap {
+
+// The spec we use for configuring the trace
+struct TraceSpec {
+  int _stack_depth;
+  bool _use_backtrace;
+  int  _only_nth;
+  bool _force;
+  bool _track_free;
+  bool _detailed_stats;
+  int _rainy_day_fund;
+
+  TraceSpec() :
+    _stack_depth(10),
+    _use_backtrace(true),
+    _only_nth(0),
+    _force(false),
+    _track_free(false),
+    _detailed_stats(false),
+    _rainy_day_fund(0) {
+  }
+};
+
+// The spec we use for configuring the dump.
+struct DumpSpec {
+  const char* _dump_file;
+  const char* _filter;
+  int         _max_entries;
+  bool        _hide_dump_allocs;
+  bool        _on_error;
+  bool        _sort_by_count;
+  int         _dump_fraction;
+  bool        _internal_stats;
+
+  DumpSpec() :
+    _dump_file(NULL),
+    _filter(NULL),
+    _max_entries(0),
+    _hide_dump_allocs(true),
+    _on_error(false),
+    _sort_by_count(false),
+    _dump_fraction(100),
+    _internal_stats(false) {
+  }
+};
```

Comment:
The header file should contain the interface to handle the trace, so I think if enabling and printing methods are available, the definition of the specs should be in the header file, since otherwise you cannot call them.